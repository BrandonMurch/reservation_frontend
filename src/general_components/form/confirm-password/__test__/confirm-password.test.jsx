import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';

import ConfirmPassword from '../index';

const fillForm = function fillForm(component) {
  let element = component.getByLabelText(/^Password/i);
  fireEvent.change(element, { target: { value: 'password' } });
  fireEvent.blur(element);
  element = component.getByLabelText(/Confirm/i);
  fireEvent.change(element, { target: { value: 'password' } });
  fireEvent.blur(element);
};

describe('<ConfirmPassword />', () => {
  let component;
  let mockFunction;
  let container = null;
  let props;

  beforeEach(() => {
    mockFunction = jest.fn();
    props = {
      updateValue: mockFunction,
    };
    component = render(<ConfirmPassword {...props} />);
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
  it('should match snapshot', () => {
    const tree = renderer.create(<ConfirmPassword {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe('password', () => {
    it('should display the password label', () => {
      const element = component.getByText(/^Password/i);
      expect(element).toBeInTheDocument();
    });

    it('should call the function on blur', () => {
      fillForm(component);
      expect(mockFunction).toHaveBeenCalledWith('', 'password');
    });
  });
  describe('confirm password', () => {
    it('should display the confirm label', () => {
      const element = component.getByText(/Confirm Password/i);
      expect(element).toBeInTheDocument();
    });
    it('should call the function on blur', () => {
      fillForm(component);
      expect(mockFunction).toHaveBeenCalledWith('password', 'password');
    });
  });
});
