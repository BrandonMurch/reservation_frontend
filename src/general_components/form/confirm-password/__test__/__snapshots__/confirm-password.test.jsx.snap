// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmPassword /> should match snapshot 1`] = `
Array [
  <div
    className="inputGroup"
  >
    <label
      className="hiddenLabelText"
      htmlFor="password"
    >
      Password
      <input
        className="input"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </label>
  </div>,
  <div
    className="inputGroup"
  >
    <label
      className="hiddenLabelText"
      htmlFor="confirm"
    >
      Confirm Password:
      <input
        className="input"
        name="confirm"
        onBlur={[Function]}
        onChange={[Function]}
        pattern=""
        placeholder="Confirm Password"
        required={true}
        type="password"
        value=""
      />
    </label>
  </div>,
]
`;
